docs/src/extend/custom-rules.md

```diff
@@ -635,73 +679,45 @@ module.exports = {
+In the preceding example, the error level is assumed to be the first argument. It is followed by the first optional argument, a string which may be either `"always"` or `"never"`. The final optional argument is an object, which may have a boolean property named `exceptRange`.
 
 To learn more about JSON Schema, we recommend looking at some examples in [website](https://json-schema.org/learn/) to start, and also reading [Understanding JSON Schema](https://json-schema.org/understanding-json-schema/) (a free ebook).
 
-**Note:** Currently you need to use full JSON Schema object rather than array in case your schema has references ($ref), because in case of array format ESLint transforms this array into a single schema without updating references that makes them incorrect (they are ignored).
-
-### Getting the Source
-
-If your rule needs to get the actual JavaScript source to work with, then use the `sourceCode.getText()` method. This method works as follows:
-
-```js
-
-// get all source
-var source = sourceCode.getText();
-
-// get source for just this AST node
-var nodeSource = sourceCode.getText(node);
-
-// get source for AST node plus previous two characters
-var nodeSourceWithPrev = sourceCode.getText(node, 2);
+TODO: i don't understand what this note is saying. could someone clarify?
 
-// get source for AST node plus following two characters
-var nodeSourceWithFollowing = sourceCode.getText(node, 0, 2);
-```
-
-In this way, you can look for patterns in the JavaScript text itself when the AST isn't providing the appropriate data (such as location of commas, semicolons, parentheses, etc.).
-
-### Accessing Comments
-
-While comments are not technically part of the AST, ESLint provides a few ways for rules to access them:
-
-#### sourceCode.getAllComments()
-
-This method returns an array of all the comments found in the program. This is useful for rules that need to check all comments regardless of location.
-
-#### sourceCode.getCommentsBefore(), sourceCode.getCommentsAfter(), and sourceCode.getCommentsInside()
-
-These methods return an array of comments that appear directly before, directly after, and inside nodes, respectively. They are useful for rules that need to check comments in relation to a given node or token.
-
-Keep in mind that the results of this method are calculated on demand.
-
-#### Token traversal methods
-
-Finally, comments can be accessed through many of `sourceCode`'s methods using the `includeComments` option.
+**Note:** Currently you need to use full JSON Schema object rather than array in case your schema has references ($ref), because in case of array format ESLint transforms this array into a single schema without updating references that makes them incorrect (they are ignored).
 
 ### Accessing Shebangs
 
+TODO: which methods to access shebang? also would be helpful to have a small example here.
```

Comment:
note to reviewers: not sure what to do with this one. please see comment in text. would appreciate your feedback.

Reply:
The latest update to the `lines-around-comment` rule includes how to check for a hashbang (along with other types of comments). It's best seen in its whole function: https://github.com/eslint/eslint/blob/8e34a04/lib/rules/lines-around-comment.js#L456-L463.

Basically you get a list of comments with `sourceCode.getAllComments()`, then in a listener for the `Program` node (or any other node), you can iterate that list and do what you need with the comments. If a comment is a hashbang, it will have `type` equal `"Shebang"` and pretty sure it will be the very first comment in that list.